### PR TITLE
♻️(project) move marion from a Django project to an app

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

As a generic application, Marion can be installed in any existing Django project as a library or Django application to generate PDF documents.

## Proposal

- [x] move `marion.documents` to a single `marion` application
- [x] move base Django project to a `sandbox`
- [x] fix static files test
- [x] fix CI configuration

### CODE REVIEW

File: sandbox/settings.py [hunk 1]
```diff
@@ -146,19 +146,19 @@ class Development(Base):
...
-    ROOT_URLCONF = "marion.urls.debug"
+    ROOT_URLCONF = "urls.debug"
 
     # Application definition
     INSTALLED_APPS = Base.INSTALLED_APPS + [
         "howard",
     ]
 
-    MARION_DOCUMENT_ISSUER_CHOICES_CLASS = "howard.defaults.DocumentIssuerChoices"
+    # MARION_DOCUMENT_ISSUER_CHOICES_CLASS = "howard.defaults.DocumentIssuerChoices"
```
Comment (jmaupetit) [inline sandbox/settings.py:156]: FIXME

File: sandbox/urls/__init__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+from django.conf import settings
+from django.conf.urls.static import static
+from django.contrib import admin
+from django.urls import include, path
+
+urlpatterns = (
+    [
+        path("admin/", admin.site.urls),
+        path("api/documents/", include("marion.urls")),
```
Comment (jmaupetit) [inline sandbox/urls/__init__.py:11]: Suggestion?

```suggestion
        path("api/marion/", include("marion.urls")),
```

File: src/marion/marion/migrations/0001_initial.py [hunk 3]
```diff
@@ -54,15 +56,17 @@ class Migration(migrations.Migration):
...
+                        ],
                         help_text="The issuer of the document among allowed ones",
                         max_length=200,
                         verbose_name="Issuer",
                     ),
                 ),
                 (
                     "context",
-                    models.JSONField(
+                    marion.models.PydanticModelField(
```
Comment (quitterie-lcs) [inline src/marion/marion/migrations/0001_initial.py:69]: Ooops...

File: sandbox/urls/__init__.py (hunk 2 above)
Comment (quitterie-lcs) [reply sandbox/urls/__init__.py:11]: It makes more sense to change with `api/marion` considering the new structure

File: src/marion/marion/tests/test_models.py [hunk 4]
```diff
@@ -168,10 +168,10 @@ def test_document_request_get_issuer_class(monkeypatch):
         models.DocumentRequest.issuer.field,
         "choices",
         [
-            ("marion.documents.issuers.Arnold", "Arnold"),
-            ("marion.documents.issuers.DummyDocument", "Dummy"),
-            ("marion.documents.issuers.Marsha", "Marsha"),
-            ("marion.documents.issuers.Richie", "Richie"),
+            ("marion.issuers.Arnold", "Arnold"),
+            ("marion.issuers.DummyDocument", "Dummy"),
+            ("marion.issuers.Marsha", "Marsha"),
```
Comment (lunika) [inline src/marion/marion/tests/test_models.py:173]: The best one.

File: src/marion/marion/urls/__init__.py [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
-    ]
-    + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
-    + static(settings.STATIC_URL, document_root=settings.STATIC_ROOT)
-)
+from rest_framework import routers
+
+from .. import views
+
+router = routers.DefaultRouter()
+router.register(r"requests", views.DocumentRequestViewSet)
```
Comment (lunika) [inline src/marion/marion/urls/__init__.py:10]: why `requests` ? I understand the needs of a prefix but why this one ?

File: src/marion/setup.cfg [hunk 6]
```diff
@@ -23,11 +23,7 @@ classifiers =
 include_package_data = True
 install_requires =
     arrow==1.0.3
-    Django==3.1.7
-    django-configurations==2.2
     djangorestframework==3.12.2
-    jsonschema==3.2.0
```
Comment (lunika) [inline src/marion/setup.cfg:29]: This one is not used at all ?

File: src/marion/marion/urls/__init__.py (hunk 5 above)
Comment (jmaupetit) [reply src/marion/marion/urls/__init__.py:10]: Originally to build an endpoint like: `/api/documents/requests/` for `DocumentRequest` objects.

File: src/marion/setup.cfg (hunk 6 above)
Comment (jmaupetit) [reply src/marion/setup.cfg:29]: Nope it should have been removed when moving from JSONSchema to Pydantic.

File: sandbox/urls/__init__.py (hunk 2 above)
Comment (jmaupetit) [reply sandbox/urls/__init__.py:11]: Yes, we finally only provide an example integration.

File: bin/pylint [hunk 7]
```diff
@@ -3,4 +3,4 @@
 # Fix docker vs local path when project sources are mounted as a volume
 params=$(echo $@ | sed "s|src/marion/||g")
 
-docker-compose run --rm marion pylint ${params}
+docker-compose run --rm -w /usr/local/src/marion marion pylint ${params}
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply bin/pylint:6]: Why not.

**Comment by jmaupetit** [comment]
@lunika & @quitterie-lcs I think I've addressed all your comments. Open for a new review?

**Review by lunika — APPROVED** [review]

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**